### PR TITLE
nixosTest: remove `machine` syntax sugar

### PR DESCRIPTION
**Pull request by roberth**
###### Motivation

The `machines` parameter is redundant and bloats the documentation.

Complicated interfaces are a plague. This simplifies one a bit.

###### Things done

- Built on platform(s): hashes will be unchanged
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/doc/manual/development/writing-nixos-tests.section.md [hunk 1]
```diff
@@ -29,17 +23,16 @@ import ./make-test-python.nix {
 
 The attribute `testScript` is a bit of Python code that executes the
 test (described below). During the test, it will start one or more
-virtual machines, the configuration of which is described by the
-attribute `machine` (if you need only one machine in your test) or by
-the attribute `nodes` (if you need multiple machines). For instance,
-[`login.nix`](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests/login.nix)
-only needs a single machine to test whether users can log in
+virtual machines, the configuration of which is described by
+the attribute `nodes` (if you need multiple machines).
```
Comment (Artturin) [inline nixos/doc/manual/development/writing-nixos-tests.section.md:27]:
```suggestion
virtual machines, the configuration of which are described by
the attribute `nodes`.
```

**Comment by SuperSandro2000** [comment]
I wouldn't mind if all the commits named like ``*: machine -> nodes.machine`` would be merged into one treewide one. 229 commits are a bit excessive.

**Comment by roberth** [comment]
I figured these would be easier to revert if necessary.

**Comment by roberth** [comment]
All good.
Any final feedback?

**Review by Artturin — APPROVED** [review]

**Review by talyz — APPROVED** [review]

**Comment by erictapen** [comment]
aa0f27abb06ca66a1dc99493ada65e2bbd6000c9 broke the `boot.biosCdrom` test:

```
$ nix build .#nixosTests.boot.biosCdrom -L
nixos-test-driver-boot-bios-cdrom> start all VLans
nixos-test-driver-boot-bios-cdrom> (finished: start all VLans, in 0.00 seconds)
nixos-test-driver-boot-bios-cdrom> additionally exposed symbols:
nixos-test-driver-boot-bios-cdrom>     ,
nixos-test-driver-boot-bios-cdrom>     ,
nixos-test-driver-boot-bios-cdrom>     start_all, test_script, machines, vlans, driver, log, os, create_machine, subtest, run_tests, join_all, retry, serial_stdout_off, serial_stdout_on, polling_condition, Machine
nixos-test-driver-boot-bios-cdrom> /nix/store/hxik5rb5m5v08ygzj1vhx3xh6g8lxy3x-nixos-test-driver-boot-bios-cdrom/test-script:1:1 undefined name 'nodes'
nixos-test-driver-boot-bios-cdrom> /nix/store/hxik5rb5m5v08ygzj1vhx3xh6g8lxy3x-nixos-test-driver-boot-bios-cdrom/test-script:8:1 undefined name 'machine'
nixos-test-driver-boot-bios-cdrom> /nix/store/hxik5rb5m5v08ygzj1vhx3xh6g8lxy3x-nixos-test-driver-boot-bios-cdrom/test-script:9:1 undefined name 'machine'
nixos-test-driver-boot-bios-cdrom> /nix/store/hxik5rb5m5v08ygzj1vhx3xh6g8lxy3x-nixos-test-driver-boot-bios-cdrom/test-script:10:1 undefined name 'machine'
nixos-test-driver-boot-bios-cdrom> /nix/store/hxik5rb5m5v08ygzj1vhx3xh6g8lxy3x-nixos-test-driver-boot-bios-cdrom/test-script:13:5 undefined name 'machine'
nixos-test-driver-boot-bios-cdrom> /nix/store/hxik5rb5m5v08ygzj1vhx3xh6g8lxy3x-nixos-test-driver-boot-bios-cdrom/test-script:14:5 undefined name 'machine'
nixos-test-driver-boot-bios-cdrom> /nix/store/hxik5rb5m5v08ygzj1vhx3xh6g8lxy3x-nixos-test-driver-boot-bios-cdrom/test-script:16:1 undefined name 'machine'
```

Any idea what is going wrong here?

**Comment by ajs124** [comment]
Should be fixed by #167048, but I'm still running tests. I skimmed the rest of aa0f27abb06ca66a1dc99493ada65e2bbd6000c9 and it seems fine.

**Comment by mweinelt** [comment]
Also broke `nixosTests.hibernate` and prevents eval of trunk-combined.

```
{UNKNOWN}: aggregate job ‘tested’ failed with the error: nixos.tests.hibernate.x86_64-linux: does not exist
 at /nix/store/vsq20sgsgsp0cs3w37wbg81vlm9k5967-hydra-perl-deps/lib/perl5/site_perl/5.34.0/Catalyst/Model/DBIC/[Schema.pm](http://schema.pm/) line 526
```

```
❯ nom-build -A nixosTests.hibernate
error: The option `machine' does not exist. Definition values:
       - In `/home/hexa/git/nixos/master/nixos/lib/build-vms.nix': <function, args: {config, lib, pkgs}>
(use '--show-trace' to show detailed location information)
```

**Comment by Artturin** [comment]
i'll submit a pr that fixes it

ofborg should really somehow catch these

**Comment by mweinelt** [comment]
Fixed in #167061.

**Pull request merged**